### PR TITLE
Add basic and API key auth to direct connections form

### LINKED ISSUE #622: Support "direct" connection via API key in webview form
Update the direct connection create form to accept API key credentials for connecting

### PR DESCRIPTION
**Pull request by noeldevelops**
## Summary of Changes

Closes #622 
- Add the ability to connect with either username/password or API key/secret for "direct" connections form
- Clean up the form - some styling, rearrange & validate fields, copy edits, etc
- Slightly smarter validation: Schema Registry URL must begin with http or https (screenshots below), neither Kafka nor SR are required, but you must fill in at least one

## Any additional details or context that should be provided?

- UX and copy edits pending team review next week
- This is still somewhat hidden with the experimental Setting while we iterate and test. To see the option - open VS Code settings and check the box for `Confluent › Experimental: Enable Direct Connections`

| Light | Dark |
| -- | -- |
| Screenshot 2024-11-26 at 2 27 39 PM | Screenshot 2024-11-26 at 2 28 09 PM |

| Schema URI Validation Error | HTML validation message if you attempt to submit incorrect pattern |
| -- | -- |
| Screenshot 2024-11-26 at 2 43 14 PM | Screenshot 2024-11-26 at 2 43 06 PM |

## Pull request checklist

Please check if your PR fulfills the following (if applicable):

##### Tests

- [ ] Added new
- [ ] Updated existing
- [ ] Deleted existing

##### Other

- [ ] All new disposables (event listeners, views, channels, etc.) collected as  for eventual cleanup?

- [ ] Does anything in this PR need to be mentioned in the user-facing [CHANGELOG](https://github.com/confluentinc/vscode/blob/main/CHANGELOG.md) or [README](https://github.com/confluentinc/vscode/blob/main/public/README.md)?
- [ ] Have you validated this change locally by [packaging](https://github.com/confluentinc/vscode/blob/main/README.md#packaging-steps) and installing the extension `.vsix` file?
  ```shell
  gulp clicktest
  ```

### CODE REVIEW

File: src/directConnect.ts [hunk 1]
```diff
@@ -43,26 +43,30 @@ export function openDirectConnectionForm(): void {
...
       body["name"],
       body["platform"],
     );
+    if (result.success) {
+      await window.showInformationMessage(`🎉 New Connection Created`, {
+        modal: true,
+        detail: `View and interact with ${body["name"]} in the Resources sidebar`,
+      });
+      directConnectForm.dispose();
+    }
```
Comment (noeldevelops) [inline src/directConnect.ts:68]: New: close the form and show a modal if we successfully added the connection

File: src/directConnect.ts [hunk 2]
```diff
@@ -43,26 +43,30 @@ export function openDirectConnectionForm(): void {
...
+    const result = await manager.createConnection(
       kafkaConfig,
       schemaRegistryConfig,
       body["name"],
       body["platform"],
     );
+    if (result.success) {
+      await window.showInformationMessage(`🎉 New Connection Created`, {
+        modal: true,
+        detail: `View and interact with ${body["name"]} in the Resources sidebar`,
```
Comment (shouples) [inline src/directConnect.ts:65]: Should we default the name to something if the user doesn't set a name? The manager uses `New Connection` if nothing was passed, but we could set something generic like `your new connection` here instead.
image

Comment (noeldevelops) [reply src/directConnect.ts:65]: ha good catch... I'll add something there as a placeholder

File: src/directConnect.ts (hunk 1 above)
Comment (shouples) [reply src/directConnect.ts]: Just personal preference speaking, but do we need this to be a modal? Not much action the user can / needs to take as a result, and we're already refreshing the Resources view on successful creation. 🤔

**Review by shouples — APPROVED** [review]
Awesome stuff! 🎉

File: src/directConnect.ts (hunk 1 above)
Comment (noeldevelops) [reply src/directConnect.ts]: Open to changing it! Let's review in the DTX meeting. I tried the non-modal info message but felt it was too subtle, and with the form closing it's almost jarring -- like you don't know what happened. I think it was for a similar reason that we switched the Scaffold form success msg to a modal as well.

Comment (shouples) [reply src/directConnect.ts]: Makes sense - in the project scaffolding scenario it seems like it's easier to lose generated files if they aren't opened in the current / a new window, but in this case the Resources view is likely still open/visible.

Another idea (for a follow-on branch) is to add a button to the notification  to specifically show/highlight it (like we do for freshly-uploaded schemas), but would need some extra logic for the "reveal" functionality in the view provider.

**Pull request merged**